crates/service_utils/src/service/types.rs

```diff
@@ -78,117 +75,93 @@ pub enum AppScope {
-                log::error!("get_app_execution_namespace: AppState not set");
-                return Err(error::ErrorInternalServerError(""));
-            }
-        };
-
-        let tenant = req.extensions().get::<Tenant>().cloned();
-        let scope = req.extensions().get::<AppScope>().cloned();
+pub struct OrganisationId(pub String);
 
-        match (
-            app_state.enable_tenant_and_scope,
-            app_state.app_env,
-            tenant,
-            scope,
-        ) {
-            (false, _, _, _) => Ok(AppExecutionNamespace("cac_v1".to_string())),
-            (true, _, Some(t), Some(_)) => Ok(AppExecutionNamespace(t.0)),
-            (true, _, None, _) => {
-                log::error!(
-                    "get_app_execution_namespace: Tenant not set in request extensions"
-                );
-                Err(error::ErrorInternalServerError(""))
-            }
-            (true, _, _, None) => {
-                log::error!(
-                    "get_app_execution_namespace: AppScope not set in request extensions"
-                );
-                Err(error::ErrorInternalServerError(""))
-            }
-        }
+impl Default for OrganisationId {
+    fn default() -> Self {
+        Self(String::from("superposition"))
     }
 }
 
-impl FromRequest for AppExecutionNamespace {
+impl FromRequest for OrganisationId {
     type Error = Error;
     type Future = Ready<Result<Self, Self::Error>>;
 
     fn from_request(
         req: &actix_web::HttpRequest,
         _: &mut actix_web::dev::Payload,
     ) -> Self::Future {
-        ready(AppExecutionNamespace::from_request_sync(req))
+        let result = req.extensions().get::<Self>().cloned().ok_or_else(|| {
+            log::error!("Organisation Id not found");
+            actix_web::error::ErrorInternalServerError("Organisation Id not found")
+        });
+
+        ready(result)
     }
 }
 
 #[derive(Deref, DerefMut, Clone, Debug)]
-pub struct Tenant(pub String);
+pub struct SchemaName(pub String);
+
+impl FromRequest for SchemaName {
+    type Error = Error;
+    type Future = Ready<Result<Self, Self::Error>>;
+
+    fn from_request(
+        req: &actix_web::HttpRequest,
+        _: &mut actix_web::dev::Payload,
+    ) -> Self::Future {
+        let result = req.extensions().get::<Self>().cloned().ok_or_else(|| {
+            log::error!("Please check that the organisation id and workspace id are being properly sent");
+            actix_web::error::ErrorInternalServerError("Please check that the organisation id and workspace id are being properly sent")
+        });
 
-impl Tenant {
-    pub fn get_tenant_name(&self) -> Result<String, String> {
-        self.deref()
-            .split('_')
-            .into_iter()
-            .last()
-            .map(str::to_string)
-            .ok_or(String::from("failed to decode tenant"))
+        ready(result)
     }
 }
-impl FromRequest for Tenant {
+
+#[derive(Clone)]
+pub struct WorkspaceRequest {
```

Comment:
```suggestion
pub struct WorkspaceContext {
```
Couldn't think of a better name :disappointed: